agent/consul/state/config_entry_intention.go

```diff
@@ -292,13 +302,15 @@ func readDestinationIntentionsFromConfigEntriesTxn(tx ReadTxn, ws memdb.WatchSet
 	names := getIntentionPrecedenceMatchServiceNames(serviceName, entMeta)
 	for _, sn := range names {
 		_, entry, err := getServiceIntentionsConfigEntryTxn(tx, ws, sn.Name, nil, &sn.EnterpriseMeta)
+		if err != nil {
+			return 0, nil, err
+		}
 		if err != nil {
 			return 0, nil, err
 		} else if entry != nil {
```

Comment:
Is this an accidental dupe?

```suggestion
		if err != nil {
			return 0, nil, err
		} else if entry != nil {
```